libraries/helpers.rb

```diff
@@ -552,3 +552,19 @@ def raise_and_write_chef_error(raise_message, chef_error = nil)
   Mixlib::ShellOut.new("echo '#{chef_error}' > /var/log/parallelcluster/chef_error_msg").run_command
   raise raise_message
 end
+
+# restart clustermgtd and slurmctld only if there are update in queue config
+def queue_config_not_update?
+  require 'yaml'
+  config = YAML.safe_load(File.read(node['cluster']['cluster_config_path']))
+  previous_config = YAML.safe_load(File.read(node['cluster']['previous_cluster_config_path']))
+  config["Scheduling"] == previous_config["Scheduling"] and compute_node_bootstrap_timeout_not_update(previous_config, config)
+end
+
+def get_compute_bootstrap_timeout(config)
```

Comment:
I'd use the word "evaluate" instead of "get" because you're not just getting a value from somewhere but you're verifying if the value is in the config otherwise setting a default value, so it is more like an "evaluation".

Reply:
Changed